src/activity_actor.cpp

```diff
@@ -1317,14 +1315,15 @@ void consume_activity_actor::finish( player_activity &act, Character & )
     if( act.id() == activity_id( "ACT_CONSUME" ) ) {
         act.set_to_null();
     }
+
     if( !temp_selections.empty() || !temp_selected_items.empty() || !temp_filter.empty() ) {
         if( act.is_null() ) {
-            player_character.assign_activity( ACT_EAT_MENU );
+            player_character.assign_activity( type );
```

Comment:
You should make a local copy of `type` since `this` may be a stale pointer at this point - see comment where all `temp_*` variables are initialized